azurelinuxagent/common/protocol/extensions_goal_state_from_extensions_config.py

```diff
@@ -131,6 +132,10 @@ def fetch_through_host():
     def id(self):
         return self._id
 
+    @property
```

Comment:
initially i created an abstraction over incarnation/etag (the "id" property), but there are some places where it is important to distinguish between them so now i am exposing them as separate properties

Reply:
in this case, shouldn't you rename the `id` property as `etag` to make it clearer?

Reply:
ohh nvm just saw that both still have the `id` property.

Reply:
Not sure if I am missing something , do we still need 'id' since we can always use 'incarnation' in case of extension_config and etag in case of vm settings ?

Also curious to know why it is important to distinguish , guess it is because of the regression you mentioned but not sure what the issue exactly is.  personally, I like the idea of abstraction since its easier to follow

Reply:
'id' is used when we just need a way to distinguish from one goal state to another (e.g. when saving to disk for debugging purposes). 'incarnation' and 'etag' are used when we actually need those values (e.g. report status needs the incarnation)

though at this point 'id' corresponds to 'incarnation' or 'etag', once the 2 goal states get merged into 1 i'm planning on making the 'id' the concatenation of both values

Reply:
Thanks for the explanation.